src/main/java/com/couchbase/lite/router/BufferOutputStream.java

```diff
@@ -0,0 +1,31 @@
+package com.couchbase.lite.router;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.LinkedList;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class BufferOutputStream extends OutputStream {
+
+    LinkedList<Byte> buffer = new LinkedList<Byte>();
```

Comment:
Using `LinkedList<Byte>` for storing data is not efficient from both memory usage and performance.